### PR TITLE
Support GHC 8.10

### PR DESCRIPTION
**Pull request by sjakobi**

### CODE REVIEW

File: HsYAML.cabal [hunk 1]
```diff
@@ -14,7 +14,7 @@ copyright:           2015-2018 Herbert Valerio Riedel
                    , 2007-2008 Oren Ben-Kiki
 category:            Text
 build-type:          Simple
-tested-with:         GHC==8.6.1, GHC==8.4.3, GHC==8.4.1, GHC==8.2.2, GHC==8.0.2, GHC==7.10.3, GHC==7.8.4, GHC==7.6.3, GHC==7.4.2
+tested-with:         GHC==8.10.1, GHC==8.6.5, GHC==8.4.4, GHC==8.2.2, GHC==8.0.2, GHC==7.10.3, GHC==7.8.4, GHC==7.6.3, GHC==7.4.2
```
Comment (sjakobi) [inline HsYAML.cabal]: If testing with GHC 8.4.1 was particularly important, I can add it back!

**Comment by sjakobi** [comment]
Note that the improved pattern match checker produces a warning:

```
src/Data/YAML/Event.hs:412:9: warning: [-Woverlapping-patterns]
    Pattern match is redundant
    In an equation for ‘go'’: go' _ _ _ xs = ...
    |
412 |         go' _ _ _ xs | False = error (show xs)
    |         ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
```

File: HsYAML.cabal (hunk 1 above)
Comment (hvr) [reply HsYAML.cabal]: It was... see #1 or #6 -- there was a screw-up in GHC 8.4.1 releases where GHC would massively violate the versioning contract in a way that was very hard to workaround... that's the reason of this particular testing point...

**Comment by hvr** [comment]
> Note that the improved pattern match checker produces a warning:

hah... I guess I need to find different tricks to leave behind debugging code which the optimizer wouldn't warn about and yet detect as dead-code during optimization :-P

**Comment by hvr** [comment]
Revised as per https://hackage.haskell.org/package/HsYAML-0.2.1.0/revisions/

File: HsYAML.cabal (hunk 1 above)
Comment (sjakobi) [reply HsYAML.cabal]: Thanks for the explanation! I re-added 8.4.1 to CI.

**Comment by hvr** [comment]
Thanks!

**Pull request merged**